### PR TITLE
FIPS: new NSS smoke test

### PR DESCRIPTION
**Pull request by jlausuch**
https://progress.opensuse.org/issues/125513

VR: https://openqa.suse.de/tests/10641679#step/nss_smoke/1

### CODE REVIEW

File: tests/fips/mozilla_nss/nss_smoke.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    if (is_transactional) {
+        trup_call("pkg install $zypper_options mozilla-nss mozilla-nss-tools");
+        process_reboot(trigger => 1);
+    } else {
+        zypper_call("in $zypper_options mozilla-nss mozilla-nss-tools");
+    }
+    record_info('mozilla-nss', script_output('rpm -q mozilla-nss'));
+    record_info('mozilla-nss', script_output('rpm -q mozilla-nss-tools'));
```
Comment (mloviska) [inline tests/fips/mozilla_nss/nss_smoke.pm:40]: First argument, s/mozilla-nss/mozilla-nss-tools/

File: tests/fips/mozilla_nss/nss_smoke.pm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    assert_script_run('cat /dev/urandom | head -n 120 > /root/seedfile.dat');
+    record_info('seedfile', script_output('cat /root/seedfile.dat'));
+    assert_script_run('echo "" > ' . $pass_file);
```
Comment (mloviska) [inline tests/fips/mozilla_nss/nss_smoke.pm:44]: is this about creating an empty file? or the quotes are required?
If it is about to create an empty file, please use `touch`

File: tests/fips/mozilla_nss/nss_smoke.pm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    my $cert_file = '/root/cert9.cer';
+    my $nssdb_dir = '/root/nssdb';
+    my $cmd = "\"CN=Daniel Duesentrieb3,O=Example Corp,L=Mountain View,ST=California,C=DE\" -d \"$nssdb_dir\" -o $cert_file -f $pass_file -z $seed_file";
+
+    if (get_var('FIPS_CERT_15SP4')) {
+        zypper_call('ar ' . get_required_var('MOZILLA_NSS_DEVEL_REPO'));
+        zypper_call("--gpg-auto-import-keys ref", 180);
+        record_info('repos', script_output('zypper lr -u'));
+        $zypper_options = "--from 1";
+    }
```
Comment (foursixnine) [inline tests/fips/mozilla_nss/nss_smoke.pm:31]: I am not a fan of this, but I guess we can talk about that tomorrow, but is better than hardcoding things

Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm:31]: me neither... but what can I do if we need specific versions? :)

File: tests/fips/mozilla_nss/nss_smoke.pm [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+sub run {
+    select_serial_terminal;
+    my $zypper_options;
+    my $pass_file = '/root/password.txt';
+    my $seed_file = '/root/seedfile.dat';
+    my $cert_file = '/root/cert9.cer';
+    my $nssdb_dir = '/root/nssdb';
+    my $cmd = "\"CN=Daniel Duesentrieb3,O=Example Corp,L=Mountain View,ST=California,C=DE\" -d \"$nssdb_dir\" -o $cert_file -f $pass_file -z $seed_file";
+
+    if (get_var('FIPS_CERT_15SP4')) {
```
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm]: done

File: tests/fips/mozilla_nss/nss_smoke.pm (hunk 3 above)
Comment (mloviska) [reply tests/fips/mozilla_nss/nss_smoke.pm]: `--from alias|name|#|URI` I guess repo alias would be better than a number

File: tests/fips/mozilla_nss/nss_smoke.pm [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        trup_call("pkg install $zypper_options mozilla-nss mozilla-nss-tools");
+        process_reboot(trigger => 1);
+    } else {
+        zypper_call("in $zypper_options mozilla-nss mozilla-nss-tools");
+    }
+    record_info('mozilla-nss', script_output('rpm -q mozilla-nss'));
+    record_info('mozilla-nss-tools', script_output('rpm -q mozilla-nss-tools'));
+
+    assert_script_run('cat /dev/urandom | head -n 120 > /root/seedfile.dat');
+    record_info('seedfile', script_output('cat /root/seedfile.dat'));
```
Comment (pdostal) [inline tests/fips/mozilla_nss/nss_smoke.pm:43]:
```suggestion
    record_info('seedfile', script_output('file /root/seedfile.dat'));
```

**Review by pdostal — APPROVED** [review]
Just nits

File: tests/fips/mozilla_nss/nss_smoke.pm (hunk 3 above)
Comment (mloviska) [reply tests/fips/mozilla_nss/nss_smoke.pm]: I mean you can export in variable only `http://download.suse.de/ibs/Devel:/Desktop:/Mozilla:/SLE-15:/next/standard/` and then give it any alias you wish when adding the repo

File: tests/fips/mozilla_nss/nss_smoke.pm [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    assert_script_run('touch ' . $pass_file);
+    assert_script_run('mkdir -p /root/nssdb');
+    assert_script_run('certutil -N -d "/root/nssdb" --empty-password');
+    assert_script_run('export NSS_FIPS=1');
```
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm:47]: Yes, but don't ask me why :D 
I got it from the command instructions.

File: tests/fips/mozilla_nss/nss_smoke.pm (hunk 5 above)
Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm:43]: why?

File: tests/fips/mozilla_nss/nss_smoke.pm [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+use utils 'zypper_call';
+
+sub run {
+    select_serial_terminal;
+    my $zypper_options;
+    my $pass_file = '/root/password.txt';
+    my $seed_file = '/root/seedfile.dat';
+    my $cert_file = '/root/cert9.cer';
+    my $nssdb_dir = '/root/nssdb';
+    my $cmd = "\"CN=Daniel Duesentrieb3,O=Example Corp,L=Mountain View,ST=California,C=DE\" -d \"$nssdb_dir\" -o $cert_file -f $pass_file -z $seed_file";
```
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm:24]: I tried this, it was my first choice... and for whatever perl reason it failed...

File: tests/fips/mozilla_nss/nss_smoke.pm (hunk 3 above)
Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm]: ok, yes, that works too

**Review by foursixnine — APPROVED** [review]

File: tests/fips/mozilla_nss/nss_smoke.pm (hunk 4 above)
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm]: Wait, I will do something to avoid this variable.

File: tests/fips/mozilla_nss/nss_smoke.pm (hunk 5 above)
Comment (pdostal) [reply tests/fips/mozilla_nss/nss_smoke.pm:43]: Too long line.

Comment (pdostal) [reply tests/fips/mozilla_nss/nss_smoke.pm:43]: It is strange to print random data.

**Comment by jlausuch** [comment]
New VR: https://openqa.suse.de/tests/10642348#

File: tests/fips/mozilla_nss/nss_smoke.pm (hunk 5 above)
Comment (jlausuch) [reply tests/fips/mozilla_nss/nss_smoke.pm:43]: ok, changed

**Pull request merged**